### PR TITLE
CHEER_UP Program!

### PR DESCRIPTION
**Pull request by AmazingDragon353**
[INPUT WANTED] [FEATURE]

## About the PR 

This adds a new program for the guardbuddies, cheer_up! It is the inverse of "heckle" and it has the buddy follow the person around and tell jokes to make them happy. It uses the same format as heckle (https://wiki.ss13.co/TermOS#GuardBuddies) to assign the buddies to this task. 

## Why's this needed? 

The buddies are adorable and need more programming options. Sometimes you don't want to insult someone, you want to make them happy and tell corny jokes!

## Sending out the bat signal

Currently there are 16 jokes, and I have more that I can think of, research, and remember, but Rome wasn't built in a day, nor by one-person, so if you have a joke, please submit it here! 

https://forms.gle/KJGhnCdPWzzBJUFCA

Thanks!

## Changelog

```changelog
(u)AmazingDragons
(*)Added a new morale-boosting problem to the mainframe for control of your favorite guardbuddies!
```

### CODE REVIEW

File: code/modules/robotics/bot/guardbot.dm [hunk 1]
```diff
@@ -3544,6 +3544,57 @@
...
+			if (..())
+				return
+
+			if (src.protected && prob(10))
+				var/buddy_cheer_up_chooser = rand(1,16)
+				var/buddy_cheer_up_answer_chooser = (buddy_cheer_up_chooser - 7) //There are more phrases than answers.
+				master.speak(buddy_cheer_up_phrases[buddy_cheer_up_chooser])
+				master.point(src.protected, 1)
+
+				if (buddy_cheer_up_chooser >= 8) //Does the phrase need an answer?
```
Comment (ZeWaka) [inline code/modules/robotics/bot/guardbot.dm:3565]: i'd rather there just be a list of one-off jokes, and then one for the two-parter starters.
so 3 total

File: code/modules/robotics/bot/guardbot.dm [hunk 2]
```diff
@@ -3544,6 +3544,57 @@
...
+					continue
+
+				var/check_name = C.name
+				if(ishuman(C) && C:wear_id)
+					check_name = C:wear_id:registered
+
+				if (ckey(check_name) == ckey(src.protected_name))
+					src.protected = C
+					buddy_is_dork = 1
+					//src.desired_emotion = GUARDING_EMOTION
```
Comment (ZeWaka) [inline code/modules/robotics/bot/guardbot.dm:3583]: ?

Comment (AmazingDragon353) [reply code/modules/robotics/bot/guardbot.dm:3585]: Could you clarify what the issue is here? I'm not super experienced with DM, and I was mostly working off of the heckle program.

Comment (ZeWaka) [reply code/modules/robotics/bot/guardbot.dm:3585]: there's commented out code

**Pull request merged**